crypto/crypto.go

```diff
@@ -144,6 +142,10 @@ func Ecrecover(hash, sig []byte) ([]byte, error) {
 // RecoverPubkey verifies the compact signature "signature" of "hash" for the
 // secp256k1 curve.
 func RecoverPubkey(signature, hash []byte) (*ecdsa.PublicKey, error) {
+	if len(hash) == 0 || types.BytesToHash(hash) == types.ZeroHash {
```

Comment:
Maybe we should compare `len` to `types.HashLength`? Or we should allow at least `> 0 && < types.HashLength` hash sizes?
I am not sure if we should allow arbitrary number of bytes?

Reply:
It makes sense, since we are using the `secp256k1` curve. I'll add the check.

Reply:
Fix: https://github.com/0xPolygon/polygon-edge/pull/1260/commits/73bb389efc7dec12005c8486505518c121910a04